pkg/kubelet/kubelet_node_status.go

```diff
@@ -474,6 +475,24 @@ func (kl *Kubelet) recordNodeSchedulableEvent(node *v1.Node) error {
 // TODO(madhusudancs): Simplify the logic for setting node conditions and
 // refactor the node status condition code out to a different file.
 func (kl *Kubelet) setNodeStatus(node *v1.Node) {
+	// TODO(dixudx): remove this in v1.14
+	// OutOfDisk condition has been removed from kubelet in 1.12
+	// For legacy node conditions, this should be removed when upgrading
+	if version.Get().Major == "1" && version.Get().Minor > "12" {
+		var conditions []v1.NodeCondition
+		var hasOutOfDisk bool
+		for i := range node.Status.Conditions {
+			if node.Status.Conditions[i].Type != v1.NodeOutOfDisk {
```

Comment:
v1.NodeOutOfDisk -> a local const.
considered that v1.NodeOutOfDisk maybe removed.

Reply:
@zhangxiaoyu-zidif `v1.NodeOutOfDisk` should be removed together in the future, e.g. v1.14.